src/timedate/timedatectl.c

```diff
@@ -700,6 +700,111 @@ static int show_timesync(int argc, char **argv, void *userdata) {
         return 0;
 }
 
+static int parse_ifindex_bus(sd_bus *bus, const char *str, int *ret) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        int r;
+
+        assert(bus);
+        assert(str);
+        assert(ret);
+
+        r = parse_ifindex(str, ret);
+        if (r >= 0)
+                return 0;
+
+        r = sd_bus_call_method(
+                        bus,
+                        "org.freedesktop.network1",
+                        "/org/freedesktop/network1",
+                        "org.freedesktop.network1.Manager",
+                        "ListLinks",
+                        &error,
+                        &reply,
+                        NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get list of interfaces: %s", bus_error_message(&error, r));
+
+        r = sd_bus_message_enter_container(reply, 'a', "(iso)");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        for (;;) {
+                const char *n;
+                int i;
+
+                r = sd_bus_message_read(reply, "(iso)", &i, &n, NULL);
+                if (r < 0)
+                        return bus_log_create_error(r);
+                if (r == 0)
+                        return log_error("Failed to find interface '%s'", str);
+
+                if (streq_ptr(n, str)) {
+                        *ret = i;
+                        return 0;;
+                }
+        }
+}
```

Comment:
hmm, why not add GetLinkByIndex() as dbus call for this?

Reply:
Implemented.